packages/@ember/-internals/glimmer/lib/renderer.ts

```diff
@@ -419,7 +421,7 @@ export abstract class Renderer {
             continue;
           }
 
-          let { shouldReflush } = root;
+          shouldReflush = root.shouldReflush;
```

Comment:
The code below removes **any** setting of `root.shouldReflush`, reading from it will always be `false` (in master this is only ever set during `runInTransaction` which this PR removes).  I think there are additional logic issues to be addressed in this refactor.

Reply:
So I was looking at that too, and it seemed to me that `shouldReflush` would actually _never_ be set to true. Like you said, it only gets set in DEBUG, and it would only get set if the value had already rendered, which would _always_ trigger an assertion: https://github.com/emberjs/ember.js/pull/18554/files#diff-1f140fcf6c460bd3aac0db8656a689b1L85

So maybe this is just legacy code that never got cleaned up?

Reply:
> So maybe this is just legacy code that never got cleaned up?

Honestly not sure, I'd have to actually poke at this on my computer (too much digging for mobile review 😛).

Reply:
Seems to not be needed, I removed it and everything appears to be working.